allensdk/brain_observatory/behavior/behavior_project_cache/behavior_ecephys_project_cache.py

```diff
@@ -0,0 +1,283 @@
+        self.cache = None
+
+        self.fetch_tries = fetch_tries
+        self.logger = logging.getLogger(self.__class__.__name__)
+
+    @property
+    def manifest(self):
+        if self.cache is None:
+            api_name = type(self.fetch_api).__name__
+            raise NotImplementedError(f"A {type(self).__name__} "
+                                      f"based on {api_name} "
+                                      "does not have an accessible manifest "
+                                      "property")
+        return self.cache.manifest
+
+    @classmethod
+    def from_s3_cache(cls, cache_dir: Union[str, Path],
+                      bucket_name: str = "visual-behavior-ecephys-data",
+                      project_name: str = "visual-behavior-ecephys"
+                      ) -> "VisualBehaviorEcephysProjectCache":
+        """instantiates this object with a connection to an s3 bucket and/or
+        a local cache related to that bucket.
+
+        Parameters
+        ----------
+        cache_dir: str or pathlib.Path
+            Path to the directory where data will be stored on the local system
+
+        bucket_name: str
+            for example, if bucket URI is 's3://mybucket' this value should be
+            'mybucket'
+
+        project_name: str
+            the name of the project this cache is supposed to access. This
+            project name is the first part of the prefix of the release data
+            objects. I.e. s3://<bucket_name>/<project_name>/<object tree>
+
+        Returns
+        -------
+        VisualBehaviorEcephysProjectCache instance
+
+        """
+        fetch_api = BehaviorEcephysProjectCloudApi.from_s3_cache(
+                cache_dir, bucket_name, project_name,
+                ui_class_name=cls.__name__)
+
+        return cls(fetch_api=fetch_api)
+
+    @classmethod
+    def from_local_cache(
+        cls,
+        cache_dir: Union[str, Path],
+        project_name: str = "visual-behavior-ecephys",
+        use_static_cache: bool = False
+    ) -> "VisualBehaviorEcephysProjectCache":
+        """instantiates this object with a local cache.
+
+        Parameters
+        ----------
+        cache_dir: str or pathlib.Path
+            Path to the directory where data will be stored on the local system
+
+        project_name: str
+            the name of the project this cache is supposed to access. This
+            project name is the first part of the prefix of the release data
+            objects. I.e. s3://<bucket_name>/<project_name>/<object tree>
+
+        Returns
+        -------
+        VisualBehaviorEcephysProjectCache instance
+
+        """
+        fetch_api = BehaviorEcephysProjectCloudApi.from_local_cache(
+            cache_dir,
+            project_name,
+            ui_class_name=cls.__name__,
+            use_static_cache=use_static_cache
+        )
+        return cls(fetch_api=fetch_api)
+
+    def _cache_not_implemented(self, method_name: str) -> None:
```

Comment:
All of the manifest methods below
```
_cache_not_implemented
load_latest_manifest
latest_downloaded_manifest_file
latest_manifest_file
load_manifest
list_all_downloaded_manifests
list_manifest_file_names
current_manifest
```
should be general enough that they should either be moved into a mixin or a base class that both `VisualBehaviorOphysProjectClass` and the current class inherit from. You can solve the problem of `isinstance(fetch_api, this_specific_class)` by using `isinstance(fetch_api, BehaviorProjectBase)`, from which both `VisualBehaviorOphysProjectClass` and this current class inherit.

Reply:
moved it into a `ProjectCacheBase` class